umake/frameworks/baseinstaller.py

```diff
@@ -237,34 +246,34 @@ def get_progress(self, progress_download, progress_requirement):
             self.last_progress_requirement = progress_requirement
 
         # we wait to have the file size to start getting progress proportion info
-
-        # try to compute balance requirement
         if self.balance_requirement_download is None:
-            if not self.pkg_to_install:
-                self.balance_requirement_download = 0
-                self.last_progress_requirement = 0
-                if self.last_progress_download is None:
-                    return
-            else:
-                # we only update if we got a progress from both sides
-                if self.last_progress_download is None or self.last_progress_requirement is None:
-                    return
-                else:
-                    # apply a minimum of 15% (no download or small download + install time)
-                    self.balance_requirement_download = max(self.pkg_size_download /
-                                                            (self.pkg_size_download + self.total_download_size),
-                                                            0.15)
-
+            if self._compute_balanced_requirement() is None:
+                return
         if not self.pbar.finished:  # drawing is delayed, so ensure we are not done first
             progress = self._calculate_progress()
             self.pbar.update(progress)
 
+    def _compute_balanced_requirement(self):
+        if not self.pkg_to_install:
+            self.balance_requirement_download = 0
+            self.last_progress_requirement = 0
+            if self.last_progress_download is None:
+                return
```

Comment:
`_compute_balanced_requirement`'s two return statements are kind of stupid way to refactor this.

I'm pretty certain I could just throw a `ReturnMainLoop` exception to stop the loop without doing the even more redundant

```python
if self._compute_balanced_requirement() is None:
    return
```